app/steps/resolveProxyHost.js

```diff
@@ -2,18 +2,20 @@
 var requestOptions = require('../../lib/requestOptions');
 
 function resolveProxyHost(container) {
-  var parsedHost;
+  var promise;
 
   if (container.options.memoizeHost && container.options.memoizedHost) {
-    parsedHost = container.options.memoizedHost;
+    promise = Promise.resolve(container.options.memoizedHost);
```

Comment:
At first glance, this appears to conflict with the goal of memoizing the host.

Reply:
I have a case where an external server defines which users should go to server A and which should go to server B.  In that case, every incoming request would query that external server.  I plan to cache the results (MRU + expiration), but I still need an async support to determine the target host url.  Does it make sense?